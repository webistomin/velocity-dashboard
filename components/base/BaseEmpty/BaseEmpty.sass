+b('base-empty')
  display: flex
  flex-direction: column
  align-items: center
  justify-content: center

  +e('img')
    margin-bottom: rem(12px)
