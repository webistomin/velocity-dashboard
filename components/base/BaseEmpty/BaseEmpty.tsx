import Vue, { CreateElement, RenderContext, VNode } from 'vue';

import './BaseEmpty.sass';

export const BaseEmpty = Vue.extend({
  functional: true,
  render(_h: CreateElement, ctx: RenderContext<{}>): VNode {
    const { staticClass, class: cls } = ctx.data;
    return (
      <div class={`base-empty ${staticClass || ''} ${cls || ''}`}>
        <v-lazy-image src='/img/empty.svg' width={184} height={152} alt='Empty box' class='base-empty__img image' />
        <strong class='base-empty__title caption'>No data was provided</strong>
      </div>
    );
  },
});
