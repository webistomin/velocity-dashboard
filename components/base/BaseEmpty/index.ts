import { BaseEmpty } from './BaseEmpty';

export default BaseEmpty;
