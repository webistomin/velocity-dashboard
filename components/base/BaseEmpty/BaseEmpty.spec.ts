import { mount } from '@vue/test-utils';
import { BaseEmpty } from './BaseEmpty';

describe('BaseEmpty', () => {
  const titleSelector = '.base-empty__title';

  const factory = (propsData = {}, options = {}) => {
    return mount(BaseEmpty, {
      propsData: {
        ...propsData,
      },
      ...options,
      stubs: ['v-lazy-image'],
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Set default text if text prop is not provided', () => {
    const wrapper = factory();
    const expectedTitle = 'No data was provided';

    const title = wrapper.find(titleSelector);
    expect(title.text()).toEqual(expectedTitle);
  });

  it('Set custom text if text prop is provided', () => {
    const expectedTitle = 'Hello world';
    const wrapper = factory({
      text: expectedTitle,
    });

    const title = wrapper.find(titleSelector);
    expect(title.text()).toEqual(expectedTitle);
  });
});
