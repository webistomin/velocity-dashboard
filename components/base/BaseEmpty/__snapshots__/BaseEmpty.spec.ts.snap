// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseEmpty Matches snapshot 1`] = `
"<div class=\\"base-empty  \\">
  <v-lazy-image-stub src=\\"/img/empty.svg\\" width=\\"184\\" height=\\"152\\" alt=\\"Empty box\\" class=\\"base-empty__img image\\"></v-lazy-image-stub><strong class=\\"base-empty__title caption\\">No data was provided</strong>
</div>"
`;
